File: packages/react-native-nitro-logger/ios/SystemLogOptions+validate.swift
import Foundation

extension SystemLogOptions {
  func validate() throws {
    guard !category.isEmpty, category.utf16.count <= 128 else {
      throw NSError(
        domain: "NitroLogger", code: 1,
        userInfo: [NSLocalizedDescriptionKey: "category must contain 1–128 UTF-16 code units"])
    }
    if let subsystem, subsystem.isEmpty || subsystem.utf16.count > 256 {
      throw NSError(
        domain: "NitroLogger", code: 2,
        userInfo: [NSLocalizedDescriptionKey: "subsystem must contain 1–256 UTF-16 code units"])
    }
  }
}

File: packages/react-native-nitro-logger/android/src/main/java/com/margelo/nitro/nitrologger/SystemLogOptions+validate.kt
package com.margelo.nitro.nitrologger

internal fun SystemLogOptions.validate() {
    require(category.isNotEmpty() && category.length <= 128) { "category must contain 1–128 UTF-16 code units" }
    val subsystem = subsystem
    require(subsystem == null || (subsystem.isNotEmpty() && subsystem.length <= 256)) { "subsystem must contain 1–256 UTF-16 code units" }
}
